### PR TITLE
Added ZXing.NET

### PR DESCRIPTION
**Pull request by YohDeadfall**
I had to rename `Capturing` to `FlashCap` since I had no will to structurize ZXing.Net files and namespaces. ZXing.Net here is a bit bigger than we have in Wasabi, but that's intentionally, I plan to remove another library used there for QR code generaton.

### CODE REVIEW

**Comment by molnard** [comment]
> I plan to remove another library used there for QR code generaton.

What is the issue with the current one?

**Comment by YohDeadfall** [comment]
ZXing.Net is battle tested .NET implementation of infamous Zebra Crossing from Java lands. I see no reason to have `WalletWasabi/Gma/QrCodeNet` when `ZXing` can do both things well.

**Comment by molnard** [comment]
> WalletWasabi/Gma/QrCodeNet

`WalletWasabi/Gma/QrCodeNet` is battle tested by us, I trust more on that - we are using that for years. What is the benefit of replacing?

**Comment by YohDeadfall** [comment]
Less binary size due to using the code shared by encoders and decoders, ability to generate geometry (requires porting to Avalonia from WPF) instead of rendering and array of `bool` which is memory and CPU inefficient.

**Comment by molnard** [comment]
Please make productive changes. The user won't notice anything about this change but it will give you and me work for another few days/weeks. 

Can we read the QR code on Mac?

**Comment by molnard** [comment]
Can you elaborate what is the goal of this PR? Maybe I didn't get the point.

**Comment by molnard** [comment]
Is this moving code here from the main repo?

**Comment by YohDeadfall** [comment]
> Can we read the QR code on Mac?

That's a thing I'm on for FlashCap. OSX has the same complex object model as Windows for camera access, only video capturing on Linux was quite easy. As soon as I make something working Kimi will be used as a tester (he has a Mac device).

> Can you elaborate what is the goal of this PR? Maybe I didn't get the point.

In short, I'm cleaning up Wasabi from external libraries incorporated into it. It started from that moment when you said that FlashCap should not be added into Wasabi directly as it's for ZXing.NET and QrCodeNet.

Since QrCodeNet is used for QR generation only and that's what ZXing.Net can do too, I decided to keep QR generation code in ZXing.Net too, so later I can drop QrCodeNet. Not immediately, but when I have some time from other important tasks.

Adding files responsible for QR generation would double the work because I have to find them later when I forget what part of ZXing.Net is responsible for. It was much easier just to trim the library checking that it compiles while removing files.

To summarize, after merging that pull request I immediately will drop `WalletWasabi.Fluent/ZXing` and it will work (already works, but I'm using a project reference at the moment for testing). And after supporting video capturing on OSX `WalletWasabi/Gma/QrCodeNet` will go away.

While writing that comment I remembered an important point about QrCodeNet. It's dead, completely dead, dead as CodePlex where it was hosted. So, in case of a potential issue supporting ZXing.Net code should be much easier.

Hope I made it clear now (:

**Comment by molnard** [comment]
> working Kimi will be used as a tester (he has a Mac device)

Adamp, Szpoti, and I also feel free to ask any of us. However, the development will be hard for you without a device. We can get you a device, reach out to Max and Rafe. 

> Adding files responsible for QR generation would double the work

It is not a goal. Cleaning up is not a goal at all. And reaching that goal will double the work.

> that moment when you said that FlashCap should not be added into Wasabi directly

That is correct. It shouldn't be.

> supporting video capturing on OSX

**That is what we are aiming for.** 

> QrCodeNet. It's dead, completely dead

Does not matter in this case. It works and QR codes won't change. 


Look I still don't see any point in prioritizing the removal of `QrCodeNet` instead of working on camera support for OSX. Focus on the goals and do not get lost in the details. @MaxHillebrand seems to agree with you so from this point you guys decide. Feel free to merge this.

**Comment by RolandUI** [comment]
IMO in the long term it would be nice to replace `QrCodeNet` because the code for making a QR image on the view side from a boolean matrix is horrible, and always cause headache when we need to maintain it. It happens maybe twice a year though.
So having native XAML would be better, but this is off-topic.

ACK for moving ZXing.NET.

**Comment by molnard** [comment]
> ACK for moving ZXing.NET.

ACK!

**Review by RolandUI — APPROVED** [review]
